File: partner-kit/examples/lookup-endpoint/dotnet/Lookup.cs
// Tiket Connect Agent — internal lookup endpoint (.NET minimal API example).
//
// The only code you write: barcode in → ticket JSON out (404 if unknown).
// Bind to localhost / your private network only — the agent is the sole caller.
//
//   dotnet run    →  http://127.0.0.1:8080/tiket/lookup?barcode=…

var builder = WebApplication.CreateBuilder(args);
builder.WebHost.UseUrls("http://127.0.0.1:8080");
var app = builder.Build();

app.MapGet("/tiket/lookup", (string barcode) =>
{
    // Replace with your real query, e.g. (EF Core / Dapper):
    //   var t = db.Tickets.SingleOrDefault(t => t.Barcode == barcode);
    //   if (t is null) return Results.NotFound();
    //   return Results.Json(new {
    //       barcode = t.Barcode, event_name = t.EventName, venue = t.VenueName,
    //       date = t.EventDate.ToString("yyyy-MM-dd"), time = t.EventTime,
    //       section = t.Section, row = t.RowNum, seat = t.SeatNum,
    //       status = t.Status, original_price = t.FaceValue,
    //       ticket_ref = t.Id.ToString(), barcode_format = "qr" });
    if (barcode == "1000000000001")
    {
        return Results.Json(new
        {
            barcode,
            event_name = "עומר אדם — סיבוב קיץ",
            venue = "היכל מנורה מבטחים",
            date = "2030-08-15",
            time = "21:00",
            section = "A", row = "12", seat = "7",
            status = "active",
            original_price = 350,
            ticket_ref = "DEMO-1",
            barcode_format = "qr",
        });
    }
    return Results.NotFound();
});

app.Run();

File: partner-kit/examples/lookup-endpoint/php/lookup.php
<?php
/**
 * Tiket Connect Agent — internal lookup endpoint (PHP example).
 *
 * The only code you write: barcode in → ticket JSON out (404 if unknown).
 * Serve it INTERNALLY only (the agent is the sole caller):
 *
 *   php -S 127.0.0.1:8080 lookup.php
 *
 * Then in tiket-agent.conf:  lookup_url = http://127.0.0.1:8080/tiket/lookup
 */

if (parse_url($_SERVER['REQUEST_URI'], PHP_URL_PATH) !== '/tiket/lookup') {
    http_response_code(404);
    exit;
}

function find_ticket(string $barcode): ?array {
    // Replace with your real query, e.g.:
    //   $stmt = $pdo->prepare('SELECT * FROM tickets WHERE barcode = ?');
    //   $stmt->execute([$barcode]);
    //   $row = $stmt->fetch(PDO::FETCH_ASSOC);
    //   if (!$row) return null;
    //   return [
    //     'barcode'        => $row['barcode'],
    //     'event_name'     => $row['event_name'],
    //     'venue'          => $row['venue_name'],
    //     'date'           => $row['event_date'],   // YYYY-MM-DD
    //     'time'           => $row['event_time'],   // HH:MM
    //     'section'        => $row['section'],
    //     'row'            => $row['row_num'],
    //     'seat'           => $row['seat_num'],
    //     'status'         => $row['status'],       // active|used|cancelled|refunded|transferred
    //     'original_price' => (float)$row['face_value'],
    //     'ticket_ref'     => $row['id'],
    //     'barcode_format' => 'qr',                 // what your gate scanners read
    //   ];
    $demo = [
        '1000000000001' => [
            'barcode' => '1000000000001',
            'event_name' => 'עומר אדם — סיבוב קיץ',
            'venue' => 'היכל מנורה מבטחים',
            'date' => '2030-08-15', 'time' => '21:00',
            'section' => 'A', 'row' => '12', 'seat' => '7',
            'status' => 'active', 'original_price' => 350,
            'ticket_ref' => 'DEMO-1', 'barcode_format' => 'qr',
        ],
    ];
    return $demo[$barcode] ?? null;
}

$ticket = find_ticket($_GET['barcode'] ?? '');
header('Content-Type: application/json; charset=utf-8');
if ($ticket === null) {
    http_response_code(404);
    echo json_encode(['error' => 'not_found']);
} else {
    echo json_encode($ticket, JSON_UNESCAPED_UNICODE);
}
